test/e2e/util/util.go

```diff
@@ -428,3 +430,78 @@ func NewIngress(name, secretName string, annotations map[string]string, dnsNames
 		},
 	}
 }
+
+func NewGateway(gatewayName, ns, secretName string, annotations map[string]string, dnsNames ...string) (*gwapiv1alpha1.Gateway, *gwapiv1alpha1.HTTPRoute) {
+	var hostnames []gwapiv1alpha1.Hostname
+	for _, dnsName := range dnsNames {
+		hostnames = append(hostnames, gwapiv1alpha1.Hostname(dnsName))
+	}
+
+	return &gwapiv1alpha1.Gateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        gatewayName,
+				Annotations: annotations,
+			},
+			Spec: gwapiv1alpha1.GatewaySpec{
+				GatewayClassName: "istio",
+				Listeners: []v1alpha1.Listener{{
+					Routes: gwapiv1alpha1.RouteBindingSelector{
+						Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
+							"gw": gatewayName,
+						}},
+					},
+					Port:     gwapiv1alpha1.PortNumber(80),
+					Hostname: (*gwapiv1alpha1.Hostname)(&dnsNames[0]),
+					TLS: &gwapiv1alpha1.GatewayTLSConfig{
+						CertificateRef: &gwapiv1alpha1.LocalObjectReference{
+							Name:  secretName,
+							Kind:  "Secret",
+							Group: "core",
+						},
+					},
+				}},
+			},
```

Comment:
Suggestion: I don't know if it makes sense for an Istio gateway, but could we test that given a Gateway with two listeners each specifying a different secret name, two `Certificate`s get created?
(See the first example in 'Examples' section in [gateway docs](https://gateway-api.sigs.k8s.io/guides/tls/))

Reply:
I can't (yet) test with an actual end-to-end test with Istio, but I added a new unit test for this case.